### PR TITLE
register user is available

### PR DESCRIPTION
**Pull request by AndreyPolezhaiev**

### CODE REVIEW

File: src/main/java/com/polezhaiev/shop/annotation/FieldMatch.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.polezhaiev.shop.annotation;
```
Comment (Rommelua) [inline src/main/java/com/polezhaiev/shop/annotation/FieldMatch.java]:
```suggestion
package com.polezhaiev.shop.validation;
```

File: src/main/java/com/polezhaiev/shop/annotation/FieldMatchValidator.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    @Override
+    public boolean isValid(Object value, final ConstraintValidatorContext context) {
+        BeanWrapper wrapper = new BeanWrapperImpl(value);
+        Object firstObj = wrapper.getPropertyValue(firstFieldName);
+        Object secondObj = wrapper.getPropertyValue(secondFieldName);
+
+        return firstObj == null && secondObj == null
+                || firstObj != null && firstObj.equals(secondObj);
```
Comment (Rommelua) [inline src/main/java/com/polezhaiev/shop/annotation/FieldMatchValidator.java]:
```suggestion
        return Objects.equals

```

File: src/main/java/com/polezhaiev/shop/dto/user/UserDto.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.polezhaiev.shop.dto.user;
+
+import lombok.Data;
+
+@Data
+public class UserDto {
```
Comment (Rommelua) [inline src/main/java/com/polezhaiev/shop/dto/user/UserDto.java]: where do you use it?

File: src/main/java/com/polezhaiev/shop/dto/user/UserRegistrationRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+@FieldMatch(first = "password", second = "repeatPassword")
+public class UserRegistrationRequestDto {
+    @NotBlank
+    @Email
+    private String email;
+    @NotBlank
+    @Length(min = 8, max = 20)
+    private String password;
+    @NotBlank
+    @Length(min = 8, max = 20)
```
Comment (Rommelua) [inline src/main/java/com/polezhaiev/shop/dto/user/UserRegistrationRequestDto.java]:
```suggestion
```

File: src/main/java/com/polezhaiev/shop/dto/user/UserRegistrationRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    private String repeatPassword;
+    @NotBlank
+    private String firstName;
+    @NotBlank
+    private String lastName;
+    @NotBlank
```
Comment (Rommelua) [inline src/main/java/com/polezhaiev/shop/dto/user/UserRegistrationRequestDto.java]:
```suggestion
```

**Review by koval-kn — APPROVED** [review]

**Pull request merged**